internals/src/read_uint.rs

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Provides functionality for reading a `usize` from an iterator of bytes.
+
+/// Reads a `usize` from an iterator.
+///
+/// `size` is the number of bytes to read, iterator is expected to return bytes in little-endian.
+///
+/// # Panics
+///
+/// If `size` is greater than 4 (we only need to read 32-bit integers).
+// We internally use implementation based on iterator so that it automatically advances as needed.
+pub fn read_uint_iter(data: &mut core::slice::Iter<'_, u8>, size: usize) -> Result<usize, EarlyEndOfScriptError> {
+    if data.len() < size {
+        return Err(EarlyEndOfScriptError);
+    };
+
+    if size > 4 {
+        panic!("read_uint_iter called with size greater 4");
+    }
+
+    let mut ret = 0;
+    for (i, item) in data.take(size).enumerate() {
+        // Size of shift left checked above - implies also that the cast is ok.
+        ret |= usize::from(*item).wrapping_shl((i * 8) as u32);
+    }
+    Ok(ret)
+}
+
+/// Indicates that we tried to read more bytes from the script than available.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
```

Comment:
These shouldn't be needed because the error is internal.

Reply:
Righto